### PR TITLE
Initial Translations Setup

### PR DESCRIPTION
**Pull request by matin-deriv**
- Simply import the hook in your component:
`import { useTranslation } from 'react-i18next';`

- Use the t function from the hook:
`const { t } = useTranslation();`

- Put the string inside the t function:
`<Layout>{t('Welcome to React')}</Layout>`

### CODE REVIEW

File: src/constants/urls.ts [hunk 1]
```diff
@@ -1,2 +1,2 @@
 export const whatsapp_url = `https://wa.me/35699578341`;
-// we can add our redirections urls here so later if we are going to change any of them we can do it just here
+// we can add our redirection urls here so later if we are going to change any of them we can do it just here
```
Comment (niloofar-deriv) [inline src/constants/urls.ts:2]: 😁

File: src/main.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import App from './App.tsx';
 import 'Styles/main.css';
+import './translations/i18n.ts';
```
Comment (niloofar-deriv) [inline src/main.tsx]:
```suggestion
import 'Translations/i18n.ts';
```

File: tsconfig.json [hunk 3]
```diff
@@ -1,37 +1,38 @@
...
+        /* Absolute path */
+        "baseUrl": "./src/",
+        "paths": {
+            "Components/*": ["components/*"],
+            "Constants/*": ["constants/*"],
+            "Contexts/*": ["contexts/*"],
+            "Hooks/*": ["hooks/*"],
+            "Utils/*": ["utils/*"],
+            "Styles/*": ["styles/*"],
+            "Translations/*": ["translations/*"]
```
Comment (niloofar-deriv) [inline tsconfig.json:33]: You need to add this to the `vite.config.ts` too

Comment (niloofar-deriv) [reply tsconfig.json:33]:
```
const ALIASES = {
    ...
    Translations: resolve(__dirname, './src/translations'),
};
```

File: src/translations/i18n.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+import { DEFAULT_LANGUAGE, LANGUAGE_KEY } from '../constants/translations';
```
Comment (niloofar-deriv) [inline src/translations/i18n.ts]:
```suggestion
import { DEFAULT_LANGUAGE, LANGUAGE_KEY } from 'Constants/translations';
```

**Review by hasan-deriv — APPROVED** [review]

**Review by mayuran-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]
LGTM 👍

**Pull request merged**